### ISSUE TITLE
[Bug]: aws_ram_principal_association causes planning failure if the association state changes to DISASSOCIATED due to external factors

### ISSUE THREAD

**Issue by ryancausey** [hashicorp/terraform-provider-aws#35296]
### Terraform Core Version

v1.5.7

### AWS Provider Version

v5.32.1

### Affected Resource(s)

* aws_ram_principal_association

### Expected Behavior

The `aws_ram_principal_association` should have seen the DISASSOCIATED state and attempted to bring the resource back to the ASSOCIATED state during the plan phase.

### Actual Behavior

The `aws_ram_principal_association` fails during the plan phase due to expected an ASSOCIATED state when there is actually a DISASSOCIATED state.

### Relevant Error/Panic Output Snippet

```shell
Planning failed. Terraform encountered an error while generating this plan.
╷
│ Error: reading RAM Resource Share (arn:aws:ram:us-west-2:12345:resource-share/asdf-asdf-asdf-asdf-asdf) Principal Association (arn:aws:organizations::12345:ou/o-asdf/ou-asdf-asdf): unexpected state 'DISASSOCIATED', wanted target 'ASSOCIATED'. last error: %!s(<nil>)
│ 
│   with aws_ram_principal_association.notifications_api_organization_access,
│   on ram.tf line 8, in resource "aws_ram_principal_association" "notifications_api_organization_access":
│    8: resource "aws_ram_principal_association" "notifications_api_organization_access" {
│ 
╵
```

### Terraform Configuration Files

```terraform
# Share our notifications source API with the rest of the Organizational Unit so it can
# be merged into an AppSync Merged API.
resource "aws_ram_resource_share" "notifications_api_organization_access" {
  name                      = "notifications_api_organization_access"
  allow_external_principals = false
}

resource "aws_ram_principal_association" "notifications_api_organization_access" {
  principal          = "arn:aws:organizations::12345:ou/o-asdf/ou-asdf-asdf"
  resource_share_arn = aws_ram_resource_share.notifications_api_organization_access.arn
}
```

### Steps to Reproduce

1. go to the AWS RAM console in the AWS Organizations management account.
2. check "Enable sharing with AWS Organizations"
3. click "save settings"
4. apply the minimal reproduction resources above while setting a valid AWS organizations organizational unit ARN as the `principal`.
5. go back to the AWS RAM console in the AWS Organizations management account.
6. uncheck "Enable sharing with AWS Organizations"
7. click "save settings"
8. run the plan again to see the failure.

### Important Factoids

I suspect that any sort of manual disassociation causes this failure, but I provided the steps that caused this to occur for me. Due to a different bug, #35295, I ended up unchecking the "Enable sharing with AWS Organizations" box via the API.

I attempted to resolve it by deleting the resource share entirely to force its re-creation, but the `aws_ram_principal_association` resource continues to fail with the same error during the plan phase.

### References

#35295

### Would you like to implement a fix?

None

**Comment by ryancausey** [comment]
I attempted to force this to work by using `terraform apply -replace=aws_ram_principal_association.notifications_api_organization_access` but the same error occurred. Using `terraform taint` also did not work. I had to use `terraform state rm` to remove the `aws_ram_principal_association` from the state entirely in order to recover. Why is the `aws_ram_principal_association` resource not set up to handle becoming DISASSOCIATED?
